test/coreneuron/unit/random/test_random.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+# =============================================================================
+# Copyright (c) 2016 - 2024 Blue Brain Project/EPFL
+#
+# See top-level LICENSE file for details.
+# =============================================================================.
+*/
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
+
+#include "coreneuron/utils/randoms/nrnran123.h"
+
+using namespace coreneuron;
+
+TEST_CASE("random123 smoke test") {
+    nrnran123_State* s;
+    int KEY_1 = 1;
+    int KEY_2 = 2;
+    int NUM_TRIES = 20;
+    double EPSILON = 0.00001;
+
+    // random123 is a fully deterministic PRNG; it should generate the following stream of doubles
+    // with (1,2) initial key:
+
+    double ref[] = {0.724128,  0.507056, 0.446231, 0.331509, 0.84732,  0.0942423, 0.5401,
+                    0.0496035, 0.286901, 0.777649, 0.584096, 0.468876, 0.545774,  0.584586,
+                    0.630361,  0.875837, 0.703636, 0.853706, 0.173552, 0.917358};
+
+    double res[NUM_TRIES];
+    s = nrnran123_newstream(KEY_1, KEY_2);
+
+    nrn_pragma_omp(target teams distribute parallel for simd map(tofrom: res[0:NUM_TRIES]) is_device_ptr(s) num_teams(1) thread_limit(1))
```

Comment:
there was a warning for `simd` clauses from NVHPC, right? If thats the case then I think we can exclude `simd` clauses. We want to parallelise on GPU and rest of the pragma should be sufficient.

Also, I believe we don't need `num_teams(1) thread_limit(1))` or `num_gangs(1) vector_length(1))` i.e. we want to parallelise loop and there is no need to limit parallelisation to serial execution?